app/src/ui/dispatcher/error-handlers.ts

```diff
@@ -359,3 +359,67 @@ export async function upstreamAlreadyExistsHandler(
 
   return null
 }
+
+/**
+ * Handler for when we attempt to checkout a branch and there are some files that would
+ * be overwritten.
+ */
+export async function localChangesOverwrittenHandler(
+  error: Error,
+  dispatcher: Dispatcher
+): Promise<Error | null> {
+  const e = asErrorWithMetadata(error)
+  if (!e) {
+    return error
+  }
+
+  const gitError = asGitError(e.underlyingError)
+  if (!gitError) {
+    return error
+  }
+
+  const dugiteError = gitError.result.gitError
+  if (!dugiteError) {
+    return error
+  }
+
+  if (dugiteError !== DugiteError.LocalChangesOverwritten) {
+    return error
+  }
+
+  const { repository, retryAction } = e.metadata
+  if (repository == null) {
+    return error
+  }
+
+  if (!(repository instanceof Repository)) {
+    return error
+  }
+
+  if (!retryAction) {
+    log.error(`No retry action provided for a git checkout error.`, e)
+    return error
+  }
+
+  // find the overwritten files from the error
+  const overwrittenFiles = []
+
+  const pathRegex = /^\t(.*)/gm
+  const { stderr } = gitError.result
+
+  let match = pathRegex.exec(stderr)
+
+  while (match) {
```

Comment:
I'm not sure how this condition is ever **not** met. From my understanding, match is of type `Array` and the `while` loop exits when some condition is `false`, I don't think an empty array will get coalesced into `false`, so it would always evaluate to true. 

Additionally, match is being set to itself every-time on L414 (identical to L410).

Reply:
It returns `null` if it doesn't match https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec, not an empty array

Reply:
> And a regex is stateful so match is not set to itself but to the next match in the stderr

That's the part I was missing!